### PR TITLE
ISPN-10296 SKIP_LISTENER_NOTIFICATION flag in hotrod

### PR DESCRIPTION
**Pull request by karesti**
https://issues.jboss.org/browse/ISPN-10296

Backport https://github.com/infinispan/infinispan/pull/7032

### CODE REVIEW

**Comment by karesti** [comment]
@wburns I know. I tested that in spring PR. Is this is enough for you ?

**Comment by wburns** [comment]
I would rather have something in the hotrod-client module as this is a specific feature it will have to support, not only for spring.

**Comment by karesti** [comment]
@wburns do you prefer I do an additional test just to test listeners are not notified?

**Comment by karesti** [comment]
ok

**Comment by wburns** [comment]
I wouldn't worry too much about the flag being set in the interceptor tbh, just adding a listener and running a method should be simpler to understand.

**Comment by karesti** [comment]
@wburns change done, backport too. let's see CI ...

**Comment by danberindei** [comment]
@karesti what's going to happen if the client is updated to use SKIP_LISTENER_NOTIFICATION but the server is an older version that doesn't know about the flag?

**Comment by karesti** [comment]
@danberindei I should check the version too?

**Comment by danberindei** [comment]
@karesti the protocol version isn't changing (at least not if you backport the flag to 9.4.x), so I don't think you can check the version.

**Comment by karesti** [comment]
@danberindei what do you suggest? I suggest doing nothing

**Comment by danberindei** [comment]
@karesti IMO you should not backport the change to 9.4.x, and both the Flag javadoc and the hotrod protocol doc should say that it's only supported in Infinispan 10.0+/HotRod 3.0+.

Is there a better option @tristantarrant?

**Comment by tristantarrant** [comment]
Please document this in the wire doc

**Comment by karesti** [comment]
@tristantarrant doc added

File: documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc [hunk 1]
```diff
@@ -805,6 +805,20 @@ even if the even is generated in a different node to where the listener is
 registered. To make them marshallable, either make them extend `Serializable`,
 `Externalizable`, or provide a custom `Externalizer` for them.
 
+==== Skiping notifications
```
Comment (oraNod) [inline documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc]: Minor typo. "Skiping" > "Skipping". Also titles should use probably use caps. "Skipping Notifications"

Comment (karesti) [reply documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc]: ok

**Comment by karesti** [comment]
changes reported

**Comment by danberindei** [comment]
@karesti still needs updates to the upgrade notes (present in the backport PR) and to the wire description (similar to https://github.com/infinispan/infinispan/blame/307bb6ebfb86f21cda24ca4b7c7b25e825656bd0/documentation/src/main/asciidoc/hotrod_protocol/hotrod_protocol.asciidoc#L1156)

**Comment by karesti** [comment]
@danberindei I added the modifications

**Comment by danberindei** [comment]
Integrated, thanks Katia!

**Pull request merged**